packages/fx-core/src/common/tools.ts

```diff
@@ -495,3 +495,35 @@ export async function copyFiles(
     });
   }
 }
+
+export function getSiteNameFromResourceId(resourceId: string): string {
+  const result = parseFromResourceId(
+    /providers\/Microsoft.Web\/sites\/([0-9a-zA-Z-]*)/,
+    resourceId
+  );
+  if (!result) {
+    throw new Error("Failed to get site name from resource id: " + resourceId);
+  }
+  return result;
+}
+
+export function getResourceGroupNameFromResourceId(resourceId: string): string {
+  const result = parseFromResourceId(/resourceGroups\/([0-9a-zA-Z-——()]*)\//, resourceId);
```

Comment:
Is the value of resourceId trusted? In case that regex of rg name would be changed in the future, can we just use reg like [^/] to loose its pattern?

Reply:
Fixed.

Reply:
Nice suggestion. Thx